### ISSUE TITLE
Question about the score attribute

### ISSUE THREAD

**Issue by gcasamat** [microsoft/EconML#322]
Hi, 
I find some strange behavior when trying to score a model. 
Here is the example (I could provide with the data if necessary):
```
model.fit(Y, T, X, W, inference = 'blb')
<econml.dml.ForestDMLCateEstimator at 0x7fd9d0699f70>
model.score(Y, T, X, W)
0.24544841969205902
model.score_
0.8528563514617827
```
Many thanks for your help!

**Comment by vsyrgkanis** [comment]
The .score_ is computed with the estimated residuals done via cross fitting.

The score(Y, T, X, W) estimates new residuals for the given Y, T, X, W by using the average of the fitted nuisance models and then calculates a score based on these residuals. 

The intend is: you should only call score(Y, T, X, W) with out-of-sample samples that were not used during training time. Otherwise you have a "train-test" spillover and overfitting takes place. In some sense by calling the score on the same data you are violating the crossfiting structure of the nuisance estimates, since the nuisances are calculated based on models that used the exact same data at training time. 

So I would use score_ if you have not set out and validation set and use score( ) to test on a validation. The score( ) on a validation also tests out-of-sample goodness of fit of the final effect model, which the score_ parameter does not, as the final stage score is calculated on the same data that the final model was trained on. So if you want to test also the goodness of fit of the final stage model too on out-of-sample, then you should set aside a validation set Y, T, X, W and call score.

**Comment by gcasamat** [comment]
Thank you for this detailed explanation. I will then use the score() function on a validation set.

**Comment by vsyrgkanis** [comment]
That model is then heavily biased, i.e. you made either parametric assumptions that were false or you over-regularized to increase stability. Both injected a lot of bias to the final model. The confidence intervals primarily capture variance (though most of our methods explicitly invoke techniques that de-bias the final estimates). However, if either the effect model is mis-specified, or you are using your own ML final model and not the defaults (e.g.  the ones in LinearDML, SparseLinearDML, ForestDML), then you can get heavily biased results.

**Comment by gcasamat** [comment]
thank you for the answer!
